operators/go/submariner_controller.go.nolint

```diff
@@ -253,3 +295,74 @@ func newPodTemplateForCR(cr *submarinerv1alpha1.Submariner) corev1.PodTemplateSp
 
 	return podTemplate
 }
+
+func newRouteAgentDaemonSet(cr *submarinerv1alpha1.Submariner) *appsv1.DaemonSet {
+	labels := map[string]string{
+		"app":       "submariner-routeagent",
+		"component": "routeagent",
+	}
+
+	matchLabels := map[string]string{
+		"app": "submariner-routeagent",
+	}
+
+	allowPrivilegeEscalation := true
+	privileged := true
+	readOnlyFileSystem := false
+	runAsNonRoot := false
+	security_context_all_cap_allow_escal := corev1.SecurityContext{
+		Capabilities:             &corev1.Capabilities{Add: []corev1.Capability{"ALL"}},
+		AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+		Privileged:               &privileged,
+		ReadOnlyRootFilesystem:   &readOnlyFileSystem,
+		RunAsNonRoot:             &runAsNonRoot,
+	}
+
+	terminationGracePeriodSeconds := int64(0)
+
+	routeAgentDaemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cr.Namespace,
+			Name:      "routeagent",
+			Labels:    labels,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{MatchLabels: matchLabels},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					TerminationGracePeriodSeconds: &terminationGracePeriodSeconds,
+					Containers: []corev1.Container{
+						{
+							Name:  "submariner-routeagent",
+							Image: "submariner-route-agent:local",
```

Comment:
We need to add the new cr.Spec.Image field, as introduced by https://github.com/dfarrell07/submariner/pull/21

Reply:
And we’ll need to revisit that slightly to distinguish engine and route agent images.

Reply:
I like the “Image” variable approach, since it gives a single value to fill in which handles all situations (other repos, local versions etc.), which makes the docs simpler ;-).

Reply:
Oh skitt, I didn't see your message, sorry

I think that using the "Image" instead of repo + version combination substracts advantages from the operator and adds more chances for human error (for example, mixing versions that don't work well together, etc..)

Reply:
On the other hand, having 2 image fields, gives you freedom to use one version of the image from somewhere, and another version of the image from devel, troubleshooting, etc.

I have seem some operators which add image override fields, to override any manual calculation, and force it, like

imageOverrides:
    engine: docker.io/myrepo/submariner
    # routeAgent:

We also need to think about how migrations will happen from A to B, for example, when we are migrating from 0.1 to 0.2, to 0.3 ... may be the operator needs to know that the target version is 0.3, but the current version (in status) is 0.1 , so it needs to execute migration steps as it progresses and migrates components in the right order, etc..

We get no simple topic ... 'XD

Reply:
OK, I wasn’t thinking of a single repo + version shared for both images, I was imagining four fields altogether (to replace the two image fields). Sharing the repo and version does make more sense by limiting the amount of rope we give the users! Migrations will need some work, yes; we’ll probably have to limit the versions then (and perhaps use a development tweak in the operator to allow us to override it).
Operator migration capabilities are described in the CSV in the package, so we get some more tweaks there; for example, we can limit the possible upgrade paths (but between versions of the operator, not versions of the images pulled by the operator).